packages/xo-server-sdn-controller/src/index.js

```diff
@@ -622,8 +641,25 @@ class SDNController extends EventEmitter {
     vni,
     encrypted,
     mtu,
+    preferredCenterId,
   }) {
-    const privateNetwork = new PrivateNetwork(this, uuidv4())
+    const preferredCenter =
+      preferredCenterId !== undefined
+        ? this._xo.getXapiObject(this._xo.getObject(preferredCenterId, 'host'))
+        : undefined
+
+    if (preferredCenter !== undefined) {
+      // Put pool of preferred center first
+      const orderedPoolIds = [preferredCenter.$pool.$id]
+      for (const poolId of poolIds) {
+        if (!orderedPoolIds.includes(poolId)) {
```

Comment:
This is not an efficient approach (`O(n²)`), you can use [`lodash.uniq`](https://devdocs.io/lodash~4/index#uniq) or simply a [`Set`](https://devdocs.io/javascript/global_objects/set):

```js
array = [...new Set(array)]
```